src/js/tech/middleware.js

```diff
@@ -0,0 +1,70 @@
+import Tech from './tech.js';
+import toTitleCase from '../utils/to-title-case.js';
+import assign from 'object.assign';
+
+const middlewares = {};
+
+export function use(type, middleware) {
+  (middlewares[type] = middlewares[type] || []).push(middleware);
+}
+
+export function setSource(src, next) {
+  setTimeout(()=>ssh(src, middlewares[src.type] || [], next, []), 1);
```

Comment:
Apparently not but yeah, I agree that there should be more whitespace. Originally added it in to verify that async works.